geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java

```diff
@@ -2397,7 +2397,8 @@ private void processInitParams(String value) {
             addExcludePattern(split[0], excludePattern);
           }
         }
-        classPatterns.add(Pattern.compile(split[0]));
+        String classPattern = split[0].replaceAll("\\$", "\\.");
```

Comment:
This change does not look safe to me. Out public contract with the ReflectionBasedAutoSerializer is that users can give it regular expressions that will be used to match class names. In a java regular expression "\$" would mean match a literal "$" character. You are changing it to say match a literal ".". I think your fix to support inner classes is at too low a level. It should have instead been in the code that creates a pattern in the precondition function.
I could be wrong about this. Lets talk about it.